fetcher.drush.inc

```diff
@@ -252,7 +255,10 @@ function fetcher_get_site($name = FALSE, $conf = array()) {
   // to the infoFetcher if no site_info.yaml was already found.
   // This only make sense if we have a name to call with.
   if ($name) {
-    $site->fetchInfo();
+    if ($site->fetchInfo() == FALSE) {
+      drush_set_error('Site does not exist');
+      exit(1);
```

Comment:
I'm still not sure we should have an exit here - this is really an API function. That said I don't want to keep putting this off and it's not like the code is super clean and well organized.